nameko/testing/utils.py

```diff
@@ -140,3 +140,65 @@ def find_free_port(host='127.0.0.1'):
     port = sock.getsockname()[1]
     sock.close()
     return port
+
+
+class ResourcePipeline(object):
+    """ Creates and destroys resources in background threads.
+
+    Creates up to `size` resources ahead of time so the caller avoids waiting
+    for lazy creation.
+    """
+
+    STOP = object()
+
+    def __init__(self, create, destroy, size=3):
+        if size == 0:
```

Comment:
so do you think 3 is still a reasonable default for other use-cases? (or do we e.g. want to force the user to choose?)